ci-operator/step-registry/upi/install/azurestack/upi-install-azurestack-commands.sh

```diff
@@ -0,0 +1,206 @@
+
+date "+%F %X" > "${SHARED_DIR}/CLUSTER_INSTALL_START_TIME"
+RESOURCE_GROUP=$(cat "${SHARED_DIR}/RESOURCE_GROUP_NAME")
+
+az group create --name "$RESOURCE_GROUP" --location "$AZURE_REGION"
+
+az storage account create -g "$RESOURCE_GROUP" --location "$AZURE_REGION" --name "${INFRA_ID}sa" --kind Storage --sku Standard_LRS
+ACCOUNT_KEY=$(az storage account keys list -g "$RESOURCE_GROUP" --account-name "${INFRA_ID}sa" --query "[0].value" -o tsv)
+
+az storage container create --name files --account-name "${INFRA_ID}sa" --public-access blob --account-key "$ACCOUNT_KEY"
+az storage blob upload --account-name "${INFRA_ID}sa" --account-key "$ACCOUNT_KEY" -c "files" -f "bootstrap.ign" -n "bootstrap.ign"
+
+AZURESTACK_UPI_LOCATION="/var/lib/openshift-install/upi/azurestack"
+az deployment group create -g "$RESOURCE_GROUP" \
+  --template-file "${AZURESTACK_UPI_LOCATION}/01_vnet.json" \
+  --parameters baseName="$INFRA_ID"
+
+
+VHD_BLOB_URL="https://rhcossa.blob.ppe3.stackpoc.com/vhd/rhcos4.vhd"
+az deployment group create -g "$RESOURCE_GROUP" \
+  --template-file "${AZURESTACK_UPI_LOCATION}/02_storage.json" \
+  --parameters vhdBlobURL="$VHD_BLOB_URL" \
+  --parameters baseName="$INFRA_ID"
+
+az deployment group create -g "$RESOURCE_GROUP" \
+  --template-file "${AZURESTACK_UPI_LOCATION}/03_infra.json" \
+  --parameters baseName="$INFRA_ID"
+
+set +euE
+
+az network dns zone create -g "$RESOURCE_GROUP" -n "${CLUSTER_NAME}.${BASE_DOMAIN}"
+PUBLIC_IP=$(az network public-ip list -g "$RESOURCE_GROUP" --query "[?name=='${INFRA_ID}-master-pip'] | [0].ipAddress" -o tsv)
+az network dns record-set a add-record -g "$RESOURCE_GROUP" -z "${CLUSTER_NAME}.${BASE_DOMAIN}" -n api -a "$PUBLIC_IP" --ttl 60
+az network dns record-set a add-record -g "$RESOURCE_GROUP" -z "${CLUSTER_NAME}.${BASE_DOMAIN}" -n api-int -a "$PUBLIC_IP" --ttl 60
+
+BOOTSTRAP_URL=$(az storage blob url --account-name "${INFRA_ID}sa" --account-key "$ACCOUNT_KEY" -c "files" -n "bootstrap.ign" -o tsv)
+BOOTSTRAP_IGNITION=$(jq -rcnM --arg v "3.1.0" --arg url "$BOOTSTRAP_URL" '{ignition:{version:$v,config:{replace:{source:$url}}}}' | base64 | tr -d '\n')
+
+az deployment group create --verbose -g "$RESOURCE_GROUP" \
+  --template-file "${AZURESTACK_UPI_LOCATION}/04_bootstrap.json" \
+  --parameters bootstrapIgnition="$BOOTSTRAP_IGNITION" \
+  --parameters sshKeyData="$SSH_KEY" \
+  --parameters baseName="$INFRA_ID" \
+  --parameters diagnosticsStorageAccountName="${INFRA_ID}sa"
+
+set -euE
+
+MASTER_IGNITION=$(cat master.ign | base64 | tr -d '\n')
+az deployment group create -g "$RESOURCE_GROUP" \
+  --template-file "${AZURESTACK_UPI_LOCATION}/05_masters.json" \
+  --parameters masterIgnition="$MASTER_IGNITION" \
+  --parameters sshKeyData="$SSH_KEY" \
+  --parameters baseName="$INFRA_ID" \
+  --parameters masterVMSize="Standard_D4_v2" \
+  --parameters diskSizeGB="1023" \
+  --parameters diagnosticsStorageAccountName="${INFRA_ID}sa"
+
```

Comment:
We also need to add: 

- wait for bootstrap
- bootstrap destroy
- worker creation
https://github.com/patrickdillon/installer/blob/azurestack-upi/upi/azurestack/run-upi.sh#L151-L168

Then CSR approval:
Perhaps we can use GCP as a guide:
https://github.com/openshift/release/blob/master/ci-operator/step-registry/upi/install/gcp/upi-install-gcp-commands.sh#L491-L500
@jstuever mentioned that the approval depends on another command in the script for exiting (because it runs as a background process, looping forever). I think it is this one which you already have, but perhaps @jstuever can confirm:
```shell
trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
```

Then need to add the *.apps DNS record:
https://github.com/openshift/installer/blob/d5bb5cf68e00f3cafa6b7ced75e47d2a697a957e/docs/user/azure/install_upi_azurestack.md#add-the-ingress-dns-records

Finally, wait for install complete, redact sensitive data, and clean up:
https://github.com/openshift/release/blob/master/ci-operator/step-registry/upi/install/gcp/upi-install-gcp-commands.sh#L528-L550